src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -151,83 +219,218 @@ inline void rpp_host_rng_xorwow_f32_initialize_multiseed_stream(RpptXorwowState
+    sincos_ps(pV, &pU, &pV);                                                                            // std::sin(v) and std::cos(v) computation
+    pRngVals[0] = _mm256_mul_ps(pU, pS);                                                                // u = std::sin(v) * s;
+    pRngVals[1] = _mm256_mul_ps(pV, pS);                                                                // v = std::cos(v) * s;
+}
+
+inline void rpp_host_rng_8_gaussian_f32_sse(__m128 *pRngVals, __m128i *pxXorwowStateX, __m128i *pxXorwowStateCounter)
+{
+    __m128 pU, pV, pS;                                                                          // Rpp32f u, v, s;
+    pU = _mm_cvtepi32_ps(rpp_host_rng_xorwow_4_u32_sse(pxXorwowStateX, pxXorwowStateCounter));  // u = (Rpp32f)rpp_host_rng_xorwow_u32(xorwowState);
+    pV = _mm_cvtepi32_ps(rpp_host_rng_xorwow_4_u32_sse(pxXorwowStateX, pxXorwowStateCounter));  // v = (Rpp32f)rpp_host_rng_xorwow_u32(xorwowState);
+    pS = _mm_cmplt_ps(pU, xmm_p0);                                                              // Adjust int32 out of bound values in float for u
+    pU = _mm_or_ps(_mm_andnot_ps(pS, pU), _mm_and_ps(pS, _mm_add_ps(xmm_p2Pow32, pU)));         // Adjust int32 out of bound values in float for u
+    pS = _mm_cmplt_ps(pV, xmm_p0);                                                              // Adjust int32 out of bound values in float for v
+    pV = _mm_or_ps(_mm_andnot_ps(pS, pV), _mm_and_ps(pS, _mm_add_ps(xmm_p2Pow32, pV)));         // Adjust int32 out of bound values in float for v
+    pU = _mm_fmadd_ps(pU, xmm_p2Pow32Inv, xmm_p2Pow32InvDiv2);                                  // u = u * RPP_2POW32_INV + RPP_2POW32_INV_DIV_2;
+    pV = _mm_fmadd_ps(pV, xmm_p2Pow32InvMul2Pi, xmm_p2Pow32InvMul2PiDiv2);                      // v = v * RPP_2POW32_INV_MUL_2PI + RPP_2POW32_INV_MUL_2PI_DIV_2;
+    pS = _mm_div_ps(xmm_p1, rpp_host_math_inverse_sqrt_4_sse(_mm_mul_ps(xmm_pm2, log_ps(pU)))); // s = 1 / rpp_host_math_inverse_sqrt_1(-2.0f * std::log(u));
+    sincos_ps(pV, &pU, &pV);                                                                    // std::sin(v) and std::cos(v) computation
+    pRngVals[0] = _mm_mul_ps(pU, pS);                                                           // u = std::sin(v) * s;
+    pRngVals[1] = _mm_mul_ps(pV, pS);                                                           // v = std::cos(v) * s;
+}
+
+inline float rpp_host_rng_1_gaussian_f32(RpptXorwowStateBoxMuller *xorwowState)
+{
+    if(xorwowState->boxMullerFlag == 0)
+    {
+        Rpp32f u, v, s;
+        u = (Rpp32f)rpp_host_rng_xorwow_u32(xorwowState) * RPP_2POW32_INV + RPP_2POW32_INV_DIV_2;
+        v = (Rpp32f)rpp_host_rng_xorwow_u32(xorwowState) * RPP_2POW32_INV_MUL_2PI + RPP_2POW32_INV_MUL_2PI_DIV_2;
+        s = 1 / rpp_host_math_inverse_sqrt_1(-2.0f * std::log(u));
```

Comment:
this is an inverse operation and will affect performance. Explore ways to avoid this

Reply:
Removed this 👍. Explored a couple options, looks like it might be better to retain inverse sqrt for hip side and just use SSE/AVX2 sqrt intrinsic for host based on the performance numbers.